testnet/stacks-node/src/tests/signer/mod.rs

```diff
@@ -355,6 +358,24 @@ impl<S: Signer<T> + Send + 'static, T: SignerEventTrait + 'static> SignerTest<Sp
         info!("Nakamoto block mine time elapsed: {mined_block_elapsed_time:?}");
     }
 
+    /// Helper function to run some code and then wait for a nakamoto block to be mined.
+    /// Note: this function does _not_ mine a BTC block.
```

Comment:
Could you add a comment in here explaining that `f` is called after recording the before state to trigger the new block.